### ISSUE TITLE
Support enabling NAU metrics in "aws_vpc" resource

### ISSUE THREAD

**Issue by tcharewicz** [terraform-aws-modules/terraform-aws-vpc#837]
## Is your request related to a new offering from AWS?
Is this functionality available in the [AWS provider for Terraform](https://registry.terraform.io/providers/hashicorp/aws/latest/docs)? See [CHANGELOG.md](https://github.com/hashicorp/terraform-provider-aws/blob/main/CHANGELOG.md), too.
  - Yes ✅: please list the AWS provider version which introduced this functionality

It will in the version 4.35.0 (Unreleased), but is not released yet.

## Is your request related to a problem? Please describe.
No

## Describe the solution you'd like.
Enable NAU metrics in "aws"vpc" resource, like in the PR to AWS provider.
https://github.com/hashicorp/terraform-provider-aws/issues/27139

```
resource "aws_vpc" "main" {
  cidr_block = "10.0.0.0/16"

  enable_network_address_usage_metrics = true   ### New flag.
}
```

## Describe alternatives you've considered.
Enable by hand from GUI and CLI.

**Comment by mkielar** [comment]
[Terraform AWS Provider v4.35.0][aws_provider] has been released with support for `enable_network_address_usage_metrics`.

[aws_provider]: https://github.com/hashicorp/terraform-provider-aws/releases/tag/v4.35.0

**Comment by mkielar** [comment]
Not pushing, just curious if there are any plans on releasing this @antonbabenko? It's a needed feature...

**Comment by mkielar** [comment]
Bad bot :P

**Comment by lorengordon** [comment]
Not stale

**Comment by lorengordon** [comment]
@antonbabenko @bryantbiggs Could someone save this from the stalebot?

**Comment by mkielar** [comment]
I guess there's no hope :(

**Comment by antonbabenko** [comment]
This issue has been resolved in version 4.0.0 :tada:
